### DISCUSSION TITLE
Smartlogger integration

### DISCUSSION THREAD

**Discussion by epper202** [wlcrs/huawei_solar#735] [Ideas]
Hi, thanks for making the repo!
I have encountered a situation where there are 4 inverters daisy chained via modbus to the smartlogger1000.
With HA I have successfully pulled data from registers via modbus TCP port 502 from the logger. It works just fine with the "smartlogger modbus interface documentation". 
 I tried to set up the smartlogger via integration to check if it was possible but it fails. Would this be a nice feature in the future?

Egbert

**Comment by epper202** [comment]
Double, didnt search properly

**Reply by Roving-Ronin** [reply]
@epper202  See existing discussion: https://github.com/wlcrs/huawei_solar/discussions/211

As per that, Thijs has no plans to add support for the SmartLogger(s). If your skilled at coding, you could add this support via the https://github.com/wlcrs/huawei-solar-lib, that this integration uses to communicate. I looked at it, and the formula you need to use to work out which modbus register ports a SmartLogger plant uses (vs the set ones defined for a non SmartLogger site) and it was beyond my software skills :-(

**Reply by epper202** [reply]
Hehe beyond my skills, I think, as well without looking at it. So far i managed to communicate with the smartlogger via de modbus tcp integration from home assistant. Reading and writing goes just fine on a 210.000kW plant 😄 

Can do it also w/o home assistant however the graphical interface brings so much for me

**Reply by Roving-Ronin** [reply]
@epper202  Do you have a Github Repository / somewhere that you have the code for SmartLogger, that you can share?
